packages/block-library/src/search/index.php

```diff
@@ -48,6 +48,8 @@ function render_block_core_search( $attributes ) {
 	// This variable is a constant and its value is always false at this moment.
 	// It is defined this way because some values depend on it, in case it changes in the future.
 	$open_by_default = false;
+	// Check if the block is using the enhanced pagination.
+	$enhanced_pagination = isset( $block->context['enhancedPagination'] ) && $block->context['enhancedPagination'];
```

Comment:
Note that this logic isn't guarded with the experiment check like in other places. In effect, this functionality automatically actives when the block is places inside Query block that has instant navigation enabled.

Reply:
Good catch as well! 👍 

Fixed in 1edf04482a631616d6ed630139ee66beb365334d